util/internal/integrationtest/common/src/main/java/org/hibernate/search/util/impl/integrationtest/common/stub/mapper/StubMapperUtils.java

```diff
@@ -20,4 +31,28 @@ public static DocumentReferenceProvider referenceProvider(String identifier) {
 	public static DocumentReferenceProvider referenceProvider(String identifier, String routingKey) {
 		return new StubDocumentReferenceProvider( identifier, routingKey );
 	}
+
+	public static <R, O> void expectLoad(ObjectLoader<R, O> objectLoaderMock,
+			Consumer<LoadingDefinitionContext<R, O>> loadingDefinition) {
+		LoadingDefinitionContext<R, O> context = new LoadingDefinitionContext<>();
+		loadingDefinition.accept( context );
+
+		EasyMock.expect( objectLoaderMock.load(
+				EasyMockUtils.collectionAnyOrderMatcher( new ArrayList<>( context.loadingMap.keySet() ) )
+		) )
+				.andAnswer(
+						() -> ( (List<R>) EasyMock.getCurrentArguments()[0] ).stream()
+						.map( context.loadingMap::get )
+						.collect( Collectors.toList() )
+				);
+	}
+
+	public static class LoadingDefinitionContext<R, O> {
```

Comment:
Maybe we can define `expectLoad` method and `LoadingDefinitionContext` class in the same project where they are used: `hibernate-search-integrationtest-backend-tck`. So that we could remove the generics R and O, improving the readability.

Reply:
I'll add a comment explaining what R and O are.